ocs_ci/ocs/pillowfight.py

```diff
@@ -70,51 +69,65 @@ def run_pillowfights(self):
         """
         ocp_local = OCP(namespace=self.namespace)
         pf_files = listdir(constants.TEMPLATE_PILLOWFIGHT_DIR)
-        for pf_yaml in pf_files:
-            pf_fullpath = join(constants.TEMPLATE_PILLOWFIGHT_DIR, pf_yaml)
-            if not pf_fullpath.endswith('.yaml'):
-                continue
-            if not isfile(pf_fullpath):
-                continue
-            pfight = templating.load_yaml(pf_fullpath)
-            lpillowfight = OCS(**pfight)
-            lpillowfight.create()
-            pf_completion_info = ''
-            pf_pod = ''
-            for pillowfight_pod in TimeoutSampler(
-                self.WAIT_FOR_TIME,
-                3,
-                get_pod_name_by_pattern,
-                'pillowfight',
-                self.COUCHBASE_OPERATOR
-            ):
-                try:
-                    pf_pod = pillowfight_pod[0]
-                    pod_info = self.up_check.exec_oc_cmd(
-                        f"get pods {pf_pod} -o json"
-                    )
-                    pf_status = pod_info['status']['containerStatuses'][0]['state']
-                    if 'terminated' in pf_status:
-                        pf_completion_info = pf_status['terminated']['reason']
+        for i in range(replicas):
+            for pf_yaml in pf_files:
+                pf_fullpath = join(constants.TEMPLATE_PILLOWFIGHT_DIR, pf_yaml)
+                if not pf_fullpath.endswith('.yaml'):
+                    continue
+                if not isfile(pf_fullpath):
+                    continue
+
+                # for basic-fillowfight.yaml
+                pfight = templating.load_yaml(pf_fullpath)
+                pfight['metadata']['name'] = 'pillowfight-rbd-simple' + f"{i}"
+
+                lpillowfight = OCS(**pfight)
+                lpillowfight.create()
+
+        pods_info = {}
+        for pillowfight_pod in TimeoutSampler(
+            self.WAIT_FOR_TIME,
+            3,
+            get_pod_name_by_pattern,
+            'pillowfight',
+            constants.COUCHBASE_OPERATOR
+        ):
+            try:
+                if len(pillowfight_pod) == replicas:
+                    counter = 0
+                    for pf_pod in pillowfight_pod:
+                        pod_info = self.up_check.exec_oc_cmd(
+                            f"get pods {pf_pod} -o json"
+                        )
+                        pf_status = pod_info['status']['containerStatuses'][0]['state']
```

Comment:
Here if you waiting for the pillowfight pod to reach completed state why dont you use wait_for_resource_state/wait_for_resource()?

Reply:
wait_for_resource_state would check for pod status 'Running'. Here couchbase pods, need to be phase: 'Running' as well as containerStatuses: state should be 'running' too. First, couchbase pods will be 'Running', to get containerStatuses: 'running' , takes little more time. So is_up_and_running() would wait till containerStatuses: 'running'